DB-8846 changes in CDH-installation.md: Java Configuration Options for HBase Master

martinrupp opened the pull request:
(first version of this only added to olapServer.classPath)

martinrupp commented:
(this has been checked on srv131 to work)

martinrupp commented:
> There is no /opt/cloudera/parcels/SPARK2 in CDH6.3.0 any more (SPARK2 is not a separate parcel but part of CDH). All those paths should be fixed.

OK! We have this line here
-Dsplice.spark.yarn.jars=/opt/cloudera/parcels/SPARK2/lib/spark2/jars/*

should this be removed completely?

OlegMazurov commented:
There are two places where SPARK2 is used. You can check with srv131 configuration to see what we actually use with CDH 6.x.

martinrupp commented:
> There are two places where SPARK2 is used. You can check with srv131 configuration to see what we actually use with CDH 6.x.

ok i checked on srv131. changes are:

/opt/cloudera/parcels/SPARK2/lib/spark2/jars/
->
/opt/cloudera/parcels/CDH/lib/spark/jars/

in two places. will amend commit

martinrupp commented:
see conversation on slack about `-XX:MaxPermSize=512`, `-Dsplice.spark.master=yarn-client` to `-Dsplice.spark.master=yarn` and `-Dspark.compaction.reserved.slots=4`

dgomezferro commented on platforms/cdh6.3.0/docs/CDH-installation.md:
````diff
@@ -550,13 +550,13 @@ com.splicemachine.hbase.BackupEndpointObserver</code>
 
    ```
-    -XX:MaxPermSize=512M -XX:+HeapDumpOnOutOfMemoryError -XX:MaxDirectMemorySize=2g -XX:+AlwaysPreTouch -XX:+UseParNewGC -XX:+UseConcMarkSweepGC -XX:CMSInitiatingOccupancyFraction=70 -XX:+CMSParallelRemarkEnabled -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.port=10101 -Dsplice.spark.enabled=true -Dsplice.spark.app.name=S ...
+   -XX:+HeapDumpOnOutOfMemoryError -XX:MaxDirectMemorySize=2g -XX:+AlwaysPreTouch -XX:+UseParNewGC -XX:+UseConcMarkSweepGC -XX:CMSInitiatingOccupancyFraction=70 -XX:+CMSParallelRemarkEnabled -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.port=10101 -Dsplice.spark.enabled=true -Dsplice.spark.app.name=SpliceMachine -Dsplice. ...
````
> -Dsplice.spark.master=yarn

This looks good but it's missing the deploy-mode, I would explicitly list it even though the default is client: `-Dsplice.spark.submit.deployMode=client`

jyuanca replied:
Please remove -Dsplice.spark.shuffle.compress=false

jyuanca replied:
Why -Dsplice.olapServer.classpath=/opt/cloudera/parcels/CDH/lib/hive/lib/*?

martinrupp replied:
> > -Dsplice.spark.master=yarn
> 
> This looks good but it's missing the deploy-mode, I would explicitly list it even though the default is client: `-Dsplice.spark.submit.deployMode=client`

done

> Please remove -Dsplice.spark.shuffle.compress=false
done

> Why -Dsplice.olapServer.classpath=/opt/cloudera/parcels/CDH/lib/hive/lib/*

i don't know about this one, this was already done like that before, and is also used e.g. on srv131

martinrupp commented:
New change does the following:

- classpath has to be added both for olapServer AND spark.executor.extraClassPath
- added `-Dsplice.spark.submit.deployMode=client`
- removed `-Dsplice.spark.shuffle.compress=false`
- `-Dsplice.spark.master=yarn-client` -> `-Dsplice.spark.master=yarn`
- changed
`/opt/cloudera/parcels/SPARK2/lib/spark2/jars/`
->
`/opt/cloudera/parcels/CDH/lib/spark/jars/`

Please see comparision srv131 vs CDH-Installation here (Sheet "v3")
https://docs.google.com/spreadsheets/d/1jqa_Qku7qyDGBHvOaPZl5mYFZMEnGeH-lGIJHr2hnU0/edit?usp=sharing#gid=1360568476

Changes are currently (here=the CDH-Installation.md config):

- `-Dsplice.spark.driver.maxResultSize`:  0 on srv131, here 2g 
- `-Dsplice.spark.dynamicAllocation.cachedExecutorIdleTimeout` and `-Dsplice.spark.dynamicAllocation.executorIdleTimeout`: 1200 on srv131, 120 here.
- `-Dsplice.spark.dynamicAllocation.minExecutors`: 3 on srv131, here 4
- `-Dsplice.spark.executor.memory`: 32500m on srv131, here 24g
- `-Dsplice.spark.maxRemoteBlockSizeFetchToMem`: 83886080 on srv131, here 105000000
- `-Dsplice.spark.reducer.maxSizeInFlight`: 400m on srv131, here 512m
- `-Dsplice.spark.ui.retainedJobs`: 150 on srv131, 100 here
- `-Dsplice.spark.ui.retainedStages`: 250 on srv131, 100 here
- `-Dsplice.spark.yarn.am.memory`: 2g on srv131, 1g here

only on srv131:
- `-Dsplice.spark.executor.memoryOverhead=4096`
- `-Dsplice.spark.yarn.containerLauncherMaxThreads=12`

only here:
- `-Dsplice.spark.local.dir=/tmp`
- `-Dsplice.spark.sql.shuffle.partitions=900`
- `-Dsplice.spark.yarn.executor.memoryOverhead=8192`

Sheet "old vs. new" shows the changes we do in this change.

martinrupp commented:
@OlegMazurov said `-Dspark.compaction.reserved.slots=4` is propably wrong, should i remove it?

martinrupp commented:
> @OlegMazurov said `-Dspark.compaction.reserved.slots=4` is propably wrong, should i remove it?

@dgomezferro said this should be removed.

martinrupp commented:
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

spliceci commented:
TEST SUCCEEDED +1

yxia92 commented:
@martinrupp Is this PR specific to master? I don't see PRs for branch-3.0 and branch-2.8. Thanks!

martinrupp commented:
> +1
> 
> Please apply these changes to release 3.0 as well.
done.

- 3.0: https://github.com/splicemachine/spliceengine/pull/3711
- 2.8: https://github.com/splicemachine/spliceengine/pull/3712

note that 2.8 contains only some of the 3.0/master changes because ORC was only a problem on CDH6.3.0. i aligned the other configurations.

martinrupp commented:
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

spliceci commented:
TEST FAILED

martinrupp commented:
com.splicemachine.hbase.SpliceRegionAdminIT.testMultipleMerge seems to be sporadic (2nd time i've seen this)

martinrupp commented:
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

spliceci commented:
TEST SUCCEEDED +1
